### PR TITLE
Revert transmission to "acquisition-based" from "modulo"

### PR DESCRIPTION
**Pull request by daniel-klein**
### Description
Reverting transmission. Performance looks VERY BAD in the benchmark, 10x more time, but a practical application in TBsim changed from 107s in main to 110s with this branch. Not sure what's going on with the benchmark (or TBsim), but otherwise happy with the changes. Because this part of the model is heavily used, we might consider number or even cython for further performance optimization.

### Checklist
- [ ] Code commented & docstrings added
- [ ] New tests were needed and have been added
- [ ] A new version number was needed & changelog has been updated
- [ ] A new PyPI version needs to be released

### CODE REVIEW

File: tests/baseline.json [hunk 1]
```diff
@@ -2,25 +2,25 @@
...
+    "sir_cum_infections": 13079.0,
+    "sis_n_susceptible": 4062.0,
+    "sis_n_infected": 6917.71568627451,
+    "sis_prevalence": 0.617184370003921,
+    "sis_new_infections": 207.57843137254903,
+    "sis_cum_infections": 21173.0,
+    "sis_rel_sus": 0.4716048846681062,
+    "n_alive": 10979.71568627451,
+    "new_deaths": 45.76470588235294,
+    "cum_deaths": 4630.0
```
Comment (cliffckerr) [inline tests/baseline.json:24]: I was surprised that the results changed so little, based on my earlier toy implementation that seemed to imply a 10% difference in the number of infections ([here](https://github.com/starsimhub/starsim/issues/507#issuecomment-2143811645)). But then a subsequent toy investigation showed no difference ([here](https://github.com/starsimhub/starsim/issues/507#issuecomment-2156405926)).

I did a more in-depth exploration here:
https://github.com/starsimhub/starsim/issues/507#issuecomment-2156515941

TLDR, even with 10,000 replicates, there is no statistically significant difference in any result between this branch and current `main`.

**Review by cliffckerr — COMMENTED** [review]
I just did a more thorough test which found that this branch and `main` seem to be statistically identical (even though a previous toy implementation I wrote showed that it should make a difference). I think this is the most confused I've ever been on a model implementation question. 🤣

**Comment by daniel-klein** [comment]
I'm curious how much performance difference there really is in practice. On TBsim, the difference between modulo and acquisition-based (roulette) is negligible. But benchmark.json shows an order of magnitude difference. Did you get to the bottom of this difference with any of your follow-up testing?

**Comment by cliffckerr** [comment]
Superseded by #546
